pages/chaoss/visualizations/contrib_importance_pie.py

```diff
@@ -0,0 +1,263 @@
+from dash import html, dcc, callback
+import dash
+from dash import dcc
+import dash_bootstrap_components as dbc
+from dash.dependencies import Input, Output, State
+import plotly.graph_objects as go
+import pandas as pd
+import logging
+from dateutil.relativedelta import *  # type: ignore
+import plotly.express as px
+from pages.utils.graph_utils import get_graph_time_values, color_seq
+from queries.contributors_query import contributors_query as ctq
+import io
+from cache_manager.cache_manager import CacheManager as cm
+from pages.utils.job_utils import nodata_graph
+import time
+import random 
+
+PAGE = "chaoss"  
+VIZ_ID = "contrib-importance-pie"  
+
+gc_contrib_importance_pie = dbc.Card(
+    [
+        dbc.CardBody(
+            [
+                html.H3(
+                    "Contributor Importance by Action Type ",
+                    className="card-title",
+                    style={"textAlign": "center"},
+                ),
+                dbc.Popover(
+                    [
+                        dbc.PopoverHeader("Graph Info:"),
+                        dbc.PopoverBody("""
+                                        Visualizes the proportion of top k contributors and\n
+                                        remaining contributors according to a specific action\n
+                                        type. Contributors who are in the top k of contributions\n 
+                                        are anonynimzed by their unique alpha-numeric Contributor\n
+                                        ID. The remaining contributors are labeled as 'other'.\n
+                                        Option to filter out contributors that have 'bot' in their\n
+                                        login username. Note: some commits have a Contributor ID\n
+                                        as none if there is no Github account associated with the\n
+                                        email used to commit.
+                                        """),
```

Comment:
Here's a slight reframe suggestion:

"For a given kind of contribution (e.g. Commit, PR, Issue), visualizes the proportional share of the top 'k' anonymous contributors, aggregating the remaining contributors as "other".

e.g. Contributor 'A' opens the most PRs of all contributors, accounting for 1/5 of all PRs. If 'k' = 1, then the chart will have one slice for Contributor 'A' accounting for 1/5 of the area, with the remaining 4/5 representing all other contributors.

Optionally, filter out contributors who have 'bot' in their login username.

Note: Some commits may have a Contributors ID as 'none' if no Github account is associated with the email that the contributor committed as."

Reply:
What about "Top {k} Contributors by {action}" where k and action are dynamically changing based on the user input ? For the pie chart, I'm not explicitly calculating bus factor - just visualizing the data as is. I like the reframe suggestion - will update `Graph Info`.

Reply:
@JamesKunstle

Reply:
@christinaexyou @cdolfi Yeah "Top {k} Contributors by Number of {action}" might be slightly clearer but we're within a razors margin of a perfect title, to me, either way, so I'm down for your preference.

Reply:
lgtm on info and graph title, will leave for @JamesKunstle